### PR TITLE
[browser][MT] enable all MT library tests on the runtime CI pipeline

### LINKED ISSUE #88765: [browser][MT] enable all MT library tests on the runtime CI pipeline
- enable all MT library tests on the runtime CI pipeline so that it's not just smoke test
 - mark ActiveIssue anything that doesn't work and create github issue for it

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/88765. This PR is testing how many failures we have after changing the MT smoke tests to full tests.
The libs that got disabled in `tests.csproj` - I was not able to define which single test is responsible for the failure, mostly because the failures are indeterministic.

- [x] When the following will pass, do `/azp run runtime-wasm` to check extra-platforms failures.

Projects containing failing/timeouting tests:

- [x] Wasm.Browser.Config.Sample
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/Wasm.Browser.Config.Sample/console
- [x] Wasm.Browser.EventPipe.Sample
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/Wasm.Browser.EventPipe.Sample/console
- [x] Microsoft.Extensions.Configuration.EnvironmentVariables.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-Microsoft.Extensions.Configuration.EnvironmentVariables.Tests/console
- [x] Microsoft.Extensions.Configuration.Functional.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-Microsoft.Extensions.Configuration.Functional.Tests/console
- [x] Microsoft.Extensions.HostFactoryResolver.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-Microsoft.Extensions.HostFactoryResolver.Tests/console
- [x] Microsoft.Extensions.Http.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-Microsoft.Extensions.Http.Tests/console
- [x] Microsoft.Extensions.Logging.Console.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-Microsoft.Extensions.Logging.Console.Tests/console
- [x] System.Collections.Concurrent.Tests (blocked the whole lib)
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Collections.Concurrent.Tests/console
- [x] System.Collections.NonGeneric.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Collections.NonGeneric.Tests/console
- [x] System.ComponentModel.TypeConverter.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.ComponentModel.TypeConverter.Tests/console
- [x] System.Composition.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Composition.Tests/console
- [x] System.Globalization.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Globalization.Tests/console
- [x] System.IO.Pipelines.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.IO.Pipelines.Tests/console
- [x] System.Linq.Parallel.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Linq.Parallel.Tests/console
- [x] System.Private.Xml.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/27d806f8-8aa9-4799-80eb-3e6f6ef2f753/workitems/WasmTestOnBrowser-System.Private.Xml.Tests/console
- [x] System.Runtime.Extensions.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Runtime.Extensions.Tests/console
- [x] System.Runtime.Serialization.Xml.ReflectionOnly.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Runtime.Serialization.Xml.ReflectionOnly.Tests/console
- [x] System.Runtime.Serialization.Xml.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Runtime.Serialization.Xml.Tests/console
- [x] System.Runtime.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Runtime.Tests/console
- [x] System.Threading.Channels.Tests (blocked the whole lib)
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Channels.Tests/console
- [x] System.Threading.Tasks.Dataflow.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Tasks.Dataflow.Tests/console (blocked the whole lib)
- [x] System.Threading.Tasks.Parallel.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Tasks.Parallel.Tests/console
- [x] System.Threading.Tasks.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Tasks.Tests/console (blocked the whole lib)
- [x] System.Threading.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Tests/console
- [x] System.Threading.Timer.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Threading.Timer.Tests/console
- [x] System.Transactions.Local.Tests
Failure log: https://helix.dot.net/api/2019-06-17/jobs/fd364e36-1e6e-4969-a94a-ccc2c2059358/workitems/WasmTestOnBrowser-System.Transactions.Local.Tests/console

### CODE REVIEW

**Comment by radical** [comment]
If while you are iterating on this, and want to run only wasm tests remove all non-browser-wasm platforms from `eng/pipelines/common/platform-matrix.yml`, and remove some of the wasm jobs from `eng/pipelines/runtime.yml`.

**Comment by ilonatommy** [comment]
Failures are not related.

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -521,14 +521,13 @@ extends:
...
-      # Build and Smoke Tests only - Wasm Threading Legs
+      # Wasm Library tests - Threading Legs
       - template: /eng/pipelines/common/templates/wasm-library-tests.yml
         parameters:
           platforms:
             - browser_wasm
-          nameSuffix: _Threading_Smoke
+          nameSuffix: _Threading
           extraBuildArgs: /p:MonoWasmBuildVariant=multithread /p:_WasmPThreadPoolSize=8 /p:AotHostArchitecture=x64 /p:AotHostOS=$(_hostedOS)
-          shouldRunSmokeOnly: true
```
Comment (pavelsavara) [inline eng/pipelines/runtime.yml:531]: Maybe it should stay smoke on default pipeline ? And full on runtime-wasm. I'm not sure if we are not low on CI resources right now.

**Review by pavelsavara — APPROVED** [review]
Thanks for doing this!

File: eng/pipelines/runtime.yml (hunk 1 above)
Comment (ilonatommy) [reply eng/pipelines/runtime.yml:531]: @radical, I think you can help with the decision

File: eng/pipelines/extra-platforms/runtime-extra-platforms-wasm.yml [hunk 2]
```diff
@@ -133,9 +117,6 @@ jobs:
       isWasmOnlyBuild: ${{ parameters.isWasmOnlyBuild }}
       # Always run for runtime-wasm because tests are not run in runtime
       alwaysRun: ${{ parameters.isWasmOnlyBuild }}
-
-      # NOTE - Since threading is experimental, we don't want to block mainline work
-      shouldContinueOnError: true
```
Comment (radical) [inline eng/pipelines/extra-platforms/runtime-extra-platforms-wasm.yml:138]: We should keep this for a couple of weeks and watch if the job stays green. If it does, then we can drop this.

File: eng/pipelines/runtime.yml (hunk 1 above)
Comment (radical) [reply eng/pipelines/runtime.yml]: Yeah, we should keep smoke tests on `runtime.yml` for now, till we at least know that the tests are stable.

**Pull request merged**